pgmpy/tests/test_base/test_DAG.py

```diff
@@ -360,6 +361,52 @@ def test_dag_fit(self):
     def tearDown(self):
         del self.graph
 
+    def test_edge_strength_basic(self):
+        """Test basic functionality and numerical values"""
+        dag = DAG([("X", "Y"), ("Z", "Y")])
+        data = pd.DataFrame({"X": [0, 1, 0, 1], "Y": [1, 3, 0, 2], "Z": [1, 1, 0, 0]})
+        strengths = dag.edge_strength(data)
+
+        # Test return type and structure
+        self.assertTrue(isinstance(strengths, dict))
+        self.assertEqual(set(strengths.keys()), {("X", "Y"), ("Z", "Y")})
+        self.assertTrue(all(isinstance(v, float) for v in strengths.values()))
+
+        # Test numerical values from docstring example
+        self.assertAlmostEqual(strengths[("X", "Y")], 0.999999754781587, places=15)
+        self.assertAlmostEqual(strengths[("Z", "Y")], 0.9999989719376587, places=15)
+
+    def test_edge_strength_specific_edge(self):
+        """Test computing strength for specific edge"""
+        dag = DAG([("X", "Y"), ("Z", "Y")])
+        data = pd.DataFrame({"X": [0, 1, 0, 1], "Y": [1, 3, 0, 2], "Z": [1, 1, 0, 0]})
+        strength_xy = dag.edge_strength(data, edges=("X", "Y"))
+
+        # Test structure
+        self.assertEqual(set(strength_xy.keys()), {("X", "Y")})
+
+        # Test numerical value from docstring
+        self.assertAlmostEqual(strength_xy[("X", "Y")], 0.999999754781587, places=15)
+
+    def test_edge_strength_empty_dag(self):
+        """Test edge case with empty DAG"""
+        empty_dag = DAG()
+        data = pd.DataFrame({"X": [0, 1, 0, 1], "Y": [1, 3, 0, 2], "Z": [1, 1, 0, 0]})
+        self.assertEqual(empty_dag.edge_strength(data), {})
+
+    def test_edge_strength_multiple_edges(self):
+        """Test computing strength for multiple specific edges"""
+        dag = DAG([("X", "Y"), ("Z", "Y")])
+        data = pd.DataFrame({"X": [0, 1, 0, 1], "Y": [1, 3, 0, 2], "Z": [1, 1, 0, 0]})
+        strengths = dag.edge_strength(data, edges=[("X", "Y"), ("Z", "Y")])
+
+        # Test structure
+        self.assertEqual(set(strengths.keys()), {("X", "Y"), ("Z", "Y")})
+
+        # Test numerical values from docstring
+        self.assertAlmostEqual(strengths[("X", "Y")], 0.999999754781587, places=15)
+        self.assertAlmostEqual(strengths[("Z", "Y")], 0.9999989719376587, places=15)
```

Comment:
Did the LinearGaussian BN tests still didn't work when comparing to the square root of strength?